### ISSUE TITLE
Question: Is it possible to specify terminal names, colours and working directories when launching Atom?

### ISSUE THREAD

**Issue by fusion809** [platformio/platformio-atom-ide-terminal#32]
Hi,

I would like to be able to launch the following Atom command:

``` bash
atom $GHUBM/*-installer
```

which for me would open [`$GHUBM/atom-installer`](https://github.com/fusion809/atom-installer), [`$GHUBM/brackets-installer`](https://github.com/fusion809/brackets-installer), [`$GHUBM/komodo-installer`](https://github.com/fusion809/komodo-installer), [`$GHUBM/lighttable-installer`](https://github.com/fusion809/lighttable-installer) and [`$GHUBM/vscode-installer`](https://github.com/fusion809/vscode-installer). I would like this package to open up terminals for each of these folders (i.e., terminals who's current working directory is the folder in question), with each terminal having a colouring scheme I can specify in the `atom $GHUBM/*-installer` command and having a name equal to the folder's name, minus the `-installer` suffix. Is there any way for me to manage this with this package?

![](http://i.imgur.com/u44El9k.png)

If what I am asking is confusing here is a screenshot showing what I want. The only difference is that while this Atom window was customized by me right-clicking each terminal tab and choosing the respective option to change the colour or name of the tab and opening each terminal to the folder I want it at, this question is whether I can open an Atom window with these settings (i.e., these terminal colours, terminal names and each terminal opened to the respective `$GHUBM/*-installer` folder in question) from the command-line using this package?  

Thanks for your time,
Brenton
##

**Comment by dmytrokyrychuk** [comment]
I don't think that you can achieve this with `platformio-ide-terminal` package alone. But you most certainly can write a package for Atom which will open/rename/change colors/go to whatever directory you need when launching Atom. If you know how to write one, of course. Also, modification of the  `platformio-ide-terminal` package is necessary, because [its API](https://github.com/platformio/platformio-atom-ide-terminal/blob/db0b08b35c585f919049889f980b9fef648cf766/lib/platformio-ide-terminal.coffee#L10-L14) is not that big, and it does not allow to do everything you will need.

**Comment by fusion809** [comment]
My knowledge of CoffeeScript is in its infancy. It's remarkable that I've managed to even create usable forks of existing Atom packages, so I'm afraid this'd be out of my reach. So I'm asking a question at discuss.atom.io to see if someone else might be interested in implementing this change. https://discuss.atom.io/t/starting-atom-with-custom-platformio-ide-terminal-tabs-opened/29742
